runtime/reconcilers/metrics_view.go

```diff
@@ -65,14 +65,22 @@ func (r *MetricsViewReconciler) Reconcile(ctx context.Context, n *runtimev1.Reso
 		return runtime.ReconcileResult{}
 	}
 
+	var modelConnector string
+	res, err := r.C.Get(ctx, &runtimev1.ResourceName{Name: mv.Spec.Table, Kind: runtime.ResourceKindModel}, false)
+	if err == nil && res.GetModel().State.ResultConnector != "" { // ignore error since dashboard can directly reference a table as well
+		modelConnector = res.GetModel().State.ResultConnector
+	} else {
+		modelConnector = mv.Spec.Connector
+	}
```

Comment:
It should only try to infer the connector if `mv.Spec.Connector == ""` (i.e. not explicitly set)

Reply:
The `mv.Spec.Connector` is never unset : https://github.com/rilldata/rill/blob/fcd9e75b450d01cc930f2e95c669c23365c91a7d/runtime/compilers/rillv1/parse_node.go#L295-L296

Reply:
Okay I think we can leverage `ConnectorInferred` to only set it if user provides it. 
For metrics_view referring to models it will be set by newly introduced logic.
For external tables it anyways fallbacks to default OLAP connector when acquiring OLAP connection.
What do you think ?

Reply:
Yeah that sounds good